manage-todos-with-realtime-agent/realtime_over_websockets/todo_utils.py

```diff
@@ -0,0 +1,85 @@
+import json
+from typing import Annotated
+
+
+def format_todo_str(data):
+    todo_str = "Current todos:\n"
+    for i, todo in enumerate(data):
+        todo_str += f"{i + 1}. {todo['task']} ({todo['status']})\n"
+    return todo_str
+
+
+def get_data():
+    try:
+        with open("todo.json", "r") as f:
+            data = json.load(f)
+    except FileNotFoundError:
+        data = []
+    return data
+
+
+def save_data(data):
+    with open("todo.json", "w") as f:
+        json.dump(data, f, indent=4)
+
+
+def add_todo(
+    id: Annotated[int, "task id"],
+    task: Annotated[str, "task description"],
+    status: Annotated[str, "task status, either done or open"],
+):
```

Comment:
Add typing for registering the function calls
```suggestion
def add_todo(
    id: Annotated[int, "task id"],
    task: Annotated[str, "task description"],
    status: Annotated[str, "task status, either done or open"],
) -> str:
```